subprojects/build-events/src/main/java/org/gradle/internal/build/event/DefaultBuildEventsListenerRegistry.java

```diff
@@ -117,9 +118,7 @@ public void onTaskCompletion(Provider<? extends OperationCompletionListener> lis
         }
 
         ForwardingBuildEventConsumer subscription = new ForwardingBuildEventConsumer(listenerProvider, executorFactory);
-        if (listenerProvider instanceof RegisteredBuildServiceProvider) {
```

Comment:
This was being done only for OperationCompletionListeners, but not for BuildOperationListeners.

The most evident symptom was failures around ProgressLoggingFixture, due to the a listener/build service leaking across the load-after-store boundary.